src/io/fasta.rs

```diff
@@ -252,12 +361,73 @@ struct IndexRecord {
 
 
 /// A sequence record returned by the FASTA index.
+#[derive(Debug, PartialEq)]
 pub struct Sequence {
     pub name: String,
     pub len: u64,
 }
 
 
+pub struct IndexedReaderIterator<'a, R: io::Read + io::Seek + 'a> {
+    reader: &'a mut IndexedReader<R>,
+    record: IndexRecord,
+    bases_left: u64,
+    line_offset: u64,
+    buf: Vec<u8>,
+    buf_idx: usize,
+    buf_len: usize,
+}
+
+
+impl<'a, R: io::Read + io::Seek + 'a> IndexedReaderIterator<'a, R> {
+    fn fill_buffer(&mut self) -> io::Result<()> {
+        assert!(self.bases_left > 0);
+
+        while self.buf_idx >= self.buf_len {
+            let bases_read = self.reader
+                .read_line(&self.record,
+                           &mut self.line_offset,
+                           self.bases_left,
+                           &mut self.buf)?;
+
+            self.buf_idx = 0;
+            self.buf_len = bases_read as usize;
+            self.bases_left -= bases_read;
+        }
+
+        Ok(())
+    }
+}
+
+
+impl<'a, R: io::Read + io::Seek + 'a> Iterator for IndexedReaderIterator<'a, R> {
+    type Item = io::Result<u8>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.buf_idx < self.buf_len {
+            let item = Some(Ok(self.buf[self.buf_idx]));
```

Comment:
This involves a lot of copying (every byte). Is there really a use case where such an iterator would be better than using `fn read`?